enhancements/20220726-vm-migration.md

```diff
@@ -0,0 +1,109 @@
+# VM Migration controller
+
+## Summary
+
+To improve adoption we need to provider users an easy way to migrate VM workloads from other cloud providers such as 
+* vsphere
+* openstack
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/2274
+
+## Motivation
+
+### Goals
+We need a way to allow users to browse their VM workloads running on vmware, openstack or harvester clusters and migrate them to current harvester cluster.
+
+Currently there is no easy way to migrate VM's from other providers to harvester.
+
+### Non-goals [optional]
+
+
+## Proposal
+
+### User Stories
+
+#### Migrate a VM from Vsphere/Openstack/Harvester cluster
+* A user should be able to define a source vsphere/openstack or harvester cluster
+* The user should be able to browse the cluster to identify a VM, and configure its network mapping to harvester VLAN definitions.
+* After a period of time the user should be able to access the original VM running on Harvester.
+* The original VM will be powered off, and it will be up to the end user to get rid of the same.
+
+#### Story 2
+
+### User Experience In Detail
+
+User needs to define the following CRDs. Eventually this can be done via a UI based flow as well.
+
+* Source Cluster definition
+```yaml:
+apiVersion: source.harvesterhci.io/v1
+kind: Vsphere
+metadata:
+    name: cluster-name
+spec:
+    endpoint: endpoint
+    credentials: secret-reference
+```
+
+* VM importjob definition
+```yaml:
+apiVersion: importjob.harvesterhci.io/v1
+kind: VirtualMachine
```

Comment:
ditto of the above group name and the CRD name can be renamed to `VirtualMachineImport`